test/doctests/src/softscope.md

```diff
@@ -0,0 +1,52 @@
+Julia 1.5's REPL softscope
+
+```
+julia> s = 0 # global
+0
+
+julia> for i = 1:10
+           t = s + i # new local `t`
+           s = t # assign global `s`
+       end
+
+julia> s # global
+55
+
+julia> @isdefined(t) # global
+false
+```
+
+```
+julia> code = """
+       s = 0 # global
+       for i = 1:10
+           t = s + i # new local `t`
+           s = t # new local `s` with warning
+       end
+       s, # global
+       @isdefined(t) # global
+       """;
+
+julia> include_string(Main, code)
+┌ Warning: Assignment to `s` in soft scope is ambiguous because a global variable by the same name exists: `s` will be treated as a new local. Disambiguate by using `local s` to suppress this warning or `global s` to assign to the existing global variable.
+└ @ string:4
+ERROR: LoadError: UndefVarError: s not defined
+```
+
+```jldoctest
+s = 0 # global
+for i = 1:10
+    t = s + i # new local `t`
+    s = t # new local `s` with warning
+end
+s, # global
+@isdefined(t) # global
+
+# output
+
+┌ Warning: Assignment to `s` in soft scope is ambiguous because a global variable by the same name exists: `s` will be treated as a new local. Disambiguate by using `local s` to suppress this warning or `global s` to assign to the existing global variable.
+└ @ ~/dev/Documenter/test/doctests/doctests.jl:3
+ERROR: UndefVarError: s not defined
```

Comment:
I guess the question is what do you actually want to have happen here? Is this usually short hand for "run this in the REPL and here's the output you should see"? or is it actually short for "run this in a script and here's the output you should see"?

Reply:
The latter. For the latter there is the `julia>` style doctests.

Reply:
Ok, then carry on, this is perfect as it is 😸